Source/ASDisplayNode+Yoga.mm

```diff
@@ -36,11 +36,15 @@ @interface ASDisplayNode (YogaInternal)
 - (ASSizeRange)_locked_constrainedSizeForLayoutPass;
 @end
 
+@interface ASLayout (YogaInternal)
+@property (nonatomic, getter=isFlattened) BOOL flattened;
+@end
+
 @implementation ASDisplayNode (Yoga)
 
 - (void)setYogaChildren:(NSArray *)yogaChildren
 {
-  for (ASDisplayNode *child in _yogaChildren) {
+  for (ASDisplayNode *child in [_yogaChildren copy]) {
```

Comment:
This is important - if this method was called previously with a non-nil _yogaChildren (not too common), then it would always crash due to mutate-while-iterating.